agent/local/state_test.go

```diff
@@ -1414,7 +1414,7 @@ func TestAgentAntiEntropy_NodeInfo(t *testing.T) {
 		}`}
 	a.Start()
 	defer a.Shutdown()
-	testrpc.WaitForLeader(t, a.RPC, "dc1")
```

Comment:
I'm not seeing any flakiness in this test using WaitForLeader, can you put it back? WaitForTestAgent adds overhead that we should avoid whenever possible.